src/lib/network_pool/indexed_pool.ml

```diff
@@ -272,8 +271,13 @@ module For_tests = struct
       let tx' = F_sequence.head_exn split_r in
       [%test_eq: Transaction_hash.User_command_with_valid_signature.t] tx tx'
     in
-    Map.iteri all_by_fee ~f:(fun ~key:fee ~data:tx_set ->
+    Map.iteri all_by_fee ~f:(fun ~key:fee_per_wu ~data:tx_set ->
         Set.iter tx_set ~f:(fun tx ->
+            let command =
+              Transaction_hash.User_command_with_valid_signature.command tx
+            in
+            let wu = User_command.weight command in
+            let fee = Currency.Fee.of_int (fee_per_wu * wu) in
```

Comment:
This could cause precision errors if the fee is not divisible by weight unit, right?

Reply:
Fixed in 9a6e6c1